src/test/java/pages/BasePage.java

```diff
@@ -4,6 +4,7 @@
 
 public abstract class BasePage {
     protected WebDriver driver;
+    protected final String ALL_CART = "//div[text()='%s']/ancestor::div[@class='cart_item']";
```

Comment:
That's not correct. You can;t move this locator to the BasePage because LognPage for example doesn;t have this element, but it's nested from BasePage still.